internal/controller/api.go

```diff
@@ -99,3 +114,88 @@ func (a *API) getDeployLog(w http.ResponseWriter, r *http.Request) {
 		log.Error(err)
 	}
 }
+
+// getMeshNodes returns a list of mesh nodes visible from the controller, and some basic readiness info.
+func (a *API) getMeshNodes(w http.ResponseWriter, r *http.Request) {
+	podInfoList := []podInfo{}
+
+	podList, err := a.clients.ListPodWithOptions(a.meshNamespace, metav1.ListOptions{
+		LabelSelector: "component==maesh-mesh",
+	})
+	if err != nil {
+		writeErrorResponse(w, fmt.Sprintf("unable to retrieve pod list: %v", err))
+		return
+	}
+
+	for _, pod := range podList.Items {
+		readiness := true
+
+		for _, status := range pod.Status.ContainerStatuses {
+			if !status.Ready {
+				// If there is a non-ready container, pod is not ready.
+				readiness = false
+				break
+			}
+		}
+
+		p := podInfo{
+			Name:  pod.Name,
+			IP:    pod.Status.PodIP,
+			Ready: readiness,
+		}
+		podInfoList = append(podInfoList, p)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if err := json.NewEncoder(w).Encode(podInfoList); err != nil {
+		log.Error(err)
+	}
+}
+
+// getMeshNodeConfiguration returns the configuration for a named pod.
+func (a *API) getMeshNodeConfiguration(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	pod, exists, err := a.clients.GetPod(a.meshNamespace, vars["node"])
+	if err != nil {
+		writeErrorResponse(w, fmt.Sprintf("unable to retrieve pod: %v", err))
+		return
+	}
+
+	if !exists {
+		writeErrorResponse(w, fmt.Sprintf("unable to find pod: %s", vars["node"]))
+		return
+	}
+
+	resp, err := http.Get(fmt.Sprintf("http://%s:8080/api/rawdata", pod.Status.PodIP))
+	if err != nil {
+		writeErrorResponse(w, fmt.Sprintf("unable to get configuration from pod: %v", err))
+		return
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		writeErrorResponse(w, fmt.Sprintf("unable to get configuration response body from pod: %v", err))
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if _, err := w.Write(body); err != nil {
+		log.Error(err)
+	}
+}
+
+func writeErrorResponse(w http.ResponseWriter, errorMessage string) {
```

Comment:
IMO this should take a `http.Status` as well as a parameter to keep the API as user friendly as possible :p Returning 500 for every error doesn't seem ideal

Reply:
Valid.